### PR TITLE
Added user model, repository and DBSet of users

### PR DESCRIPTION
**Pull request by damjan21**

### CODE REVIEW

File: Data.Access.Layer/Repositories/UnitOfWork.cs [hunk 1]
```diff
@@ -7,9 +7,12 @@ public class UnitOfWork : IUnitOfWork, IDisposable
     {
         private readonly SqlDataContext _context;
 
-        public UnitOfWork(SqlDataContext context)
+        public IUserRepository Users { get; private set; }
+
+        public UnitOfWork(SqlDataContext context, IUserRepository users)
         {
             _context = context;
+            Users = new UserRepository(context);
```
Comment (petar-kovac) [inline Data.Access.Layer/Repositories/UnitOfWork.cs:15]: Why is this initialized here? Should this be injected thru DI?

File: Data.Access.Layer/Repositories/UserRepository.cs [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Data.Access.Layer.Repositories
+{
+    public class UserRepository : GenericRepository<User>, IUserRepository 
+    {
+        private readonly SqlDataContext _context;
+        //a
```
Comment (petar-kovac) [inline Data.Access.Layer/Repositories/UserRepository.cs:15]: Please remove the comment.

**Comment by petar-kovac** [comment]
@damjan21 please change the target branch to be "dev" instead of "main".

**Comment by damjan21** [comment]
User repo injected through DI in UnitOfWork.
IGenericRepository inherits IDisposable, GenericRepositroy imeplements IDisposable.
Added Scoped pair IUserRepositroy,UserRepository

**Comment by damjan21** [comment]
We have no dev branch.

File: Data.Access.Layer/Models/User.cs [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+
+        public string? Name { get; set; }
+
+        public string? Lastname { get; set; }
+
+        public string? Email { get; set; }
+
+        public string? Password { get; set; }
+
+        public Enums.UserRole Role { get; set; }
```
Comment (petar-kovac) [inline Data.Access.Layer/Models/User.cs:21]: Would these pieces of information be enough for a single user?  
How would you handle deleted users? What would happen if we needed to soft delete the user?
Have you check with client do you need to have information about user changes?

**Comment by damjan21** [comment]
Removed scoped pair, it will conflict with feature/genericrepository branch in the future

**Comment by petar-kovac** [comment]
Dev brach is added, you can now set it :)

**Pull request merged**